ci/build_wheel.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+package_name=$1
+package_dir=$2
+
+# Use gha-tools rapids-pip-wheel-version to generate wheel version then
+# update the necessary files
+RAPIDS_EPOCH_TIMESTAMP=$(date +%s)
+versioneer_override="$(rapids-pip-wheel-version ${RAPIDS_EPOCH_TIMESTAMP})"
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+
+bash ci/release/apply_wheel_modifications.sh ${versioneer_override} "-${RAPIDS_PY_CUDA_SUFFIX}"
+echo "The package name and/or version was modified in the package source. The git diff is:"
+git diff
+
+cd "${package_dir}"
+
+# sccache configuration
+if [ "${CI:-false}" = "false" ]; then
+  # Configure sccache for read-only mode since no credentials
+  # are available in local builds.
+  export SCCACHE_S3_NO_CREDENTIALS=true
+  export PARALLEL_LEVEL=${PARALLEL_LEVEL:-$(nproc)}
+  export RAPIDS_BUILD_TYPE=${RAPIDS_BUILD_TYPE:-"pull-request"}
+fi
+export SCCACHE_S3_KEY_PREFIX="libraft-$(arch)"
+
+# Set up for pip installation of dependencies from the nightly index
+export PIP_EXTRA_INDEX_URL=https://pypi.k8s.rapids.ai/simple
```

Comment:
When we switch to the new index we should consider upstreaming this setting to the containers. Alternatively, we could upstream it now and just update the containers later. I'm fine with either approach. The latter is probably nicer for making this PR be an exemplar of future work. In either approach we may have a small hiccup during the switch, but I think that's fine.

Reply:
Yes, we can just upstream it now and export this variable to the environment. That way, when we do change the index we won't have to change every downstream repository.

Reply:
I think that makes sense, yes.